platform/jobs/edxPlatformBokChoyMaster.groovy

```diff
@@ -0,0 +1,178 @@
+package devops
+
+import hudson.model.Build
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_WORKER
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_PARSE_SECRET
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_HIPCHAT
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_BASE_URL
+
+/*
+Example secret YAML file used by this script
+publicJobConfig:
+    open : true/false
+    jobName : name-of-jenkins-job-to-be
+    url: github-url-segment
+    url-clone : github-url-segment-clone
```

Comment:
the url and credential clone are to check out the build flow scripts from the repo, the url-clone, credential-clone, and cloneReference refer to the platform. I will change the variables names, at the time I couldn't think of a better naming scheme.